src/Symfony/Component/Console/Helper/QuestionHelper.php

```diff
@@ -44,7 +44,13 @@ public function ask(InputInterface $input, OutputInterface $output, Question $qu
             $output = $output->getErrorOutput();
         }
 
-        if (!$input->isInteractive()) {
+        $isInteractive = !$input->isInteractive();
+
+        if ($isInteractive && $question instanceof ChoiceQuestion) {
```

Comment:
Or I think that you should just put `if ($question instanceof ChoiceQuestion) {` within the existing `if (!$input->isInteractive()) {`.

Reply:
shouldn’t the `ChoiceQuestion` itself take care of returning the correct default value ?

Reply:
I don't know if we should nest theses ifs, IMO its clearer like this.

@mathroc I wanted to it like this firstly but it's hard since the getDefault is called without the choices being changed.

Reply:
Nvm @sroze you are right, it's avoid one test for nothing.